lib/pure/asynchttpserver.nim

```diff
@@ -289,6 +290,9 @@ proc processRequest(
     request.client.close()
     return false
 
+const
+  keepaliveTimeout* {.intdefine.} = 60 ### default value for keepaliveTimeout in seconds
+
```

Comment:
Great, but I think this should be stored in `AsyncHttpServer` along with other options.

Reply:
> This is a good patch when it comes to keep-alive timeout (and it might be good to keep this feature in its own PR and create a new one for the issue you're trying to fix). But you shouldn't just stop client processing when number of active descriptors is too high, that will just lead to more confusion.
> 
> You are fixing the keep-alive:off case, right? So why not check for this header and only process one request in that case?

@dom96 can you please view the PR now? I made several changes and now it doesn't close the connection.
This fix two cases "keepalive" and "connection". I did a test with the wrk and the server doesn't die.